tensorflow/core/ops/math_ops.cc

```diff
@@ -131,90 +131,102 @@ tf.complex_abs(x) ==> [5.25594902, 6.60492229]
 
 // Declares cwise unary operations signature: 't -> 't
 #define UNARY()                      \
-  Input("x: T").Output("y: T").Attr( \
-      "T: {half, float, double, int32, complex64, int64}")
+  Input("x: T").Output("y: T")
```

Comment:
@girving two quick questions for you:

- UNARY: {half, float, double, int32, complex64, int64}
- UNARY_ALL: {half, float, double, uint8, int8, int16, int32, int64, complex64}

Something like this? I am not sure UNARY_ALL is necessary.

- UNARY_REAL: {half, float, double}
- UNARY_COMPLEX: {half, float, double, complex64, ~~complex128~~}

I don't see `complex128` used in math_ops or cwise_op_\*? Delete `complex128` or add it in the MACRO and cwise_op_\*?

Thanks!